File: leetcode/c++/easy/_1550_three_consecutive_odds.cpp
// Leetcode Problem 1550

#include <gtest/gtest.h>

using namespace std;

class Solution {
public:
    bool threeConsecutiveOdds(vector<int>& arr) {
        int count_odd = 0;

        for (int n : arr) {
            if (n % 2 == 1) {
                count_odd++;
                if (count_odd == 3) {
                    return true;
                }
            } else {
                count_odd = 0;
            }
        }
        return false;
    }
};


class TestThreeConsecutiveOdds : public ::testing::Test {
protected:
    Solution solution;
};

TEST_F(TestThreeConsecutiveOdds, SimpleTestCase1) {
    vector<int> arr = {2, 6, 4, 1};

    EXPECT_FALSE(solution.threeConsecutiveOdds(arr));
}

TEST_F(TestThreeConsecutiveOdds, SimpleTestCase2) {
    vector<int> arr = {1, 2, 34, 3, 4, 5, 7, 23, 12};

    EXPECT_TRUE(solution.threeConsecutiveOdds(arr));
}

File: leetcode/rust/problems/src/easy/_1550_three_consecutive_odds.rs
// Leetcode Problem 1550

struct Solution;

#[allow(dead_code)]
impl Solution {
    pub fn three_consecutive_odds(arr: Vec<i32>) -> bool {
        let mut count_odd = 0;
        for n in &arr {
            if n % 2 == 1 {
                count_odd += 1;
                if count_odd == 3 {
                    return true;
                }
            } else {
                count_odd = 0;
            }
        }
        false
    }
}


#[test]
fn test_three_consecutive_odds() {
    let arr1 = vec![2, 6, 4, 1];
    let arr2 = vec![1, 2, 34, 3, 4, 5, 7, 23, 12];

    assert!(!Solution::three_consecutive_odds(arr1));
    assert!(Solution::three_consecutive_odds(arr2));
}